network/wsPeer.go

```diff
@@ -1115,6 +1114,9 @@ func (wp *wsPeer) sendMessagesOfInterest(messagesOfInterestGeneration uint32, me
 }
 
 func (wp *wsPeer) OnClose(f func()) {
+	wp.closersMu.Lock()
```

Comment:
Not directly related to this PR, but its' odd to me that the same type has both a `Close()` and `OnClose()` with minimal overlap.

Reply:
I get it's a handler interface.